ls1intum/Artemis
# [Enhancement] Remove Jsoup from Jenkins Job Manipulation

### Checklist

### Motivation and Context


Artemis uses JSoup to parse and manipulate the configuration files of Jenkins jobs. Since JSoup is not recommended for manipulating XML, errors and bug can occur such as #3499  and #3484. 

### Description

This PR removes JSoup from the classes that manipulate Jenkins jobs and instead uses the native parser from Java. This PR should resolve #3499.

### Steps for Testing

This PR should be tested on TS2.

1. Log in to Artemis
2. Navigate to Course Administration
3. Generate a new C programming exercise -> should work
4. Participate in it -> should work and build results should display

### Test Coverage



JenkinsService.java 88%
JenkinsJobPermissionsService.java 71%
JenkinsJobPermissionsUtils.java 94%
JenkinsJobService.java 63%
XmlFileUtils.java 69%
